### PR TITLE
Create uyuni common libs

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Extract spacewalk-backend-libs into an own package and join it again with spacewalk-usix.
This new package is named uyuni-common-libs and is required to build for python2 and 3
as it is used in the client tools repositories also on older distros.

With this change spacewalk-backend become a package which is only needed on the Server or Proxy and can be turned into a python3 only source.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **internal**

- [x] **DONE**

## Test coverage
- No tests: **already covered**

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/9475

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql" 
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests" 
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: utils/spacewalk-utils.spec [hunk 1]
```diff
@@ -95,9 +94,9 @@ Requires:       salt
 Requires:       spacewalk-admin
 Requires:       spacewalk-backend
 %if 0%{?suse_version} >= 1320
```
Comment (juliogonzalez) [inline utils/spacewalk-utils.spec]: We already now that RHEL/CentOS8 is to be python3 based, maybe we should use this occasion to add it here as well?

File: uyuni/common-libs/common/__init__.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2009--2017 Red Hat, Inc.
+# Copyright (c) 2019 SUSE LLC
```
Comment (juliogonzalez) [inline uyuni/common-libs/common/__init__.py]: I think we should keep both copyrights (in the end this is no a full rewrite).

File: uyuni/common-libs/common/__init__.py [hunk 3]
```diff
@@ -8,7 +8,6 @@
 # along with this software; if not, see
 # http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
 #
-# Red Hat trademarks are not licensed under GPLv2. No permission is
```
Comment (juliogonzalez) [inline uyuni/common-libs/common/__init__.py]: And if so, the note as well.

File: uyuni/common-libs/common/__init__.py (hunk 2 above)
Comment (mcalmer) [reply uyuni/common-libs/common/__init__.py]: For me this is a new file. And this file has just the copyright header. I think this is ok. There is no code in this file.

File: uyuni/common-libs/common/__init__.py (hunk 3 above)
Comment (mcalmer) [reply uyuni/common-libs/common/__init__.py]: Usix was dropped completly.

**Comment by mcalmer** [comment]
@juliogonzalez btw. Currently head testsuite run with this code.

File: uyuni/common-libs/common/__init__.py (hunk 2 above)
Comment (juliogonzalez) [reply uyuni/common-libs/common/__init__.py]: Ah, right. This is only `__init__.py`.

**Comment by mcalmer** [comment]
> LGTM. As long as we can verify at Head/Uyuni that this doesn't break stuff, we can merge.
> 
> Did you already add the packages to the product definitions at Uyuni as well?

Nothing changed in uyuni yet and also head still needs changes in the product.
Thats something for Monday.

**Comment by juliogonzalez** [comment]
Good, no problem. Let's discuss on Monday how we want to test this.

**Pull request merged**